testing/src/test/java/io/stargate/it/cql/TokenAwareTest.java

```diff
@@ -0,0 +1,47 @@
+package io.stargate.it.cql;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.ProtocolVersion;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.metadata.TokenMap;
+import com.datastax.oss.driver.api.core.type.codec.TypeCodecs;
+import io.stargate.it.BaseOsgiIntegrationTest;
+import io.stargate.it.driver.CqlSessionExtension;
+import io.stargate.it.driver.TestKeyspace;
+import io.stargate.it.storage.StargateEnvironmentInfo;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(CqlSessionExtension.class)
+public class TokenAwareTest extends BaseOsgiIntegrationTest {
+  @Test
+  @DisplayName("Should use all Stargate addresses when using token-aware load balancing")
+  public void testTokenMapDistribution(
+      CqlSession session, StargateEnvironmentInfo stargate, @TestKeyspace CqlIdentifier keyspace) {
+    assertThat(session.getMetadata().getTokenMap()).isPresent();
+    final TokenMap tokenMap = session.getMetadata().getTokenMap().get();
+
+    Collection<Node> expectedReplicasTried = session.getMetadata().getNodes().values();
```

Comment:
Does this test use multiple Stargate nodes?

Reply:
No, good catch! I meant to do that.

Reply:
Fixed this.